src/core/Akka.Streams/Implementation/Sinks.cs

```diff
@@ -984,7 +994,7 @@ private sealed class SubSource : SubSourceOutlet<TIn>
             {
                 private readonly Logic _logic;
                 private readonly LazySink<TIn, TMat> _stage;
-                private bool _completed;
+                internal bool Completed;
 
                 public SubSource(Logic logic, TIn firstElement) : base(logic, "LazySink")
```

Comment:
you get the `Logic` as parameter and can use `_logic.Completed` in the two places below.